Source/SSH.js

```diff
@@ -138,6 +139,8 @@ export default class SSH {
             ))
           } else reject(err)
         })
+      }).then(function(){
+        SFTP.end();
```

Comment:
eh, yeah. Well it forces every new `put` call to open up another stream, which isn't great. I've fixed that issue in my fork, along with some tests and whatever.

And, actually my fork has kind of mushroomed but you can take a look at the SSH.js file to see my fix proposal.